### PR TITLE
Add comments and correct a typo in BlockRepository

### PR DESCRIPTION
**Pull request by hungdang1312**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs [hunk 1]
```diff
@@ -12,26 +12,67 @@
...
     {
+        /// <summary>
+        /// Initializes the blockchain storage and ensure the genesis block has been created in the database.
+        /// </summary>
         Task InitializeAsync();
 
+        /// <summary>
+        /// Persist the next block hash and insert new blocks into the database.
+        /// </summary>
+        /// <param name="nextBlockHash">next block hash</param>
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: We normally start with upper case and end with a `.`
>Next block hash.

Comment (hungdang1312) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: Thanks @dangershony  I will check the Commenting guides again and fix the issues.

File: src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs [hunk 2]
```diff
@@ -12,26 +12,67 @@
...
+        /// <summary>
+        /// Retreive the transaction information asynchronously using transaction id.
+        /// </summary>
+        /// <param name="trxid">The transaction id to find.</param>
         Task<Transaction> GetTrxAsync(uint256 trxid);
 
-        Task DeleteAsync(uint256 newlockHash, List<uint256> hashes);
-
+        /// <summary>
+        /// Wipe out blocks and their transactions then replace with a new block.
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: >Wipe out blocks and their transactions then replace with a new block.
more accurate is 
>Wipe out blocks and their transactions then set the tip to the be the newBlockHash.

File: src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs [hunk 3]
```diff
@@ -12,26 +12,67 @@
...
+        /// <summary>
+        /// Get the corresponding block hash by using transaction hash.
+        /// </summary>
+        /// <param name="trxid">The transaction hash</param>
         Task<uint256> GetTrxBlockIdAsync(uint256 trxid);
 
+        /// <summary>
+        /// Set the next block hash and persist it in the database.
+        /// </summary>
+        /// <param name="nextBlockHash">The next block hash.</param>
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: You missed 'SetTxIndexAsync'

File: src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs [hunk 4]
```diff
@@ -88,6 +129,9 @@ public virtual BlockStoreRepositoryPerformanceCounter PerformanceCounterFactory(
             return new BlockStoreRepositoryPerformanceCounter(this.dateTimeProvider);
         }
 
+        /// <summary>
+        /// Initializes the blockchain storage and ensure the genesis block has been created in the database.
+        /// </summary>
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: On the implementation we use ` /// <inheritdoc />`

**Comment by dangershony** [comment]
Thanks for the contribution @hungdang1312 I will push your PR and if you want to fix based on my comments please submit a new PR.

**Pull request merged**